src/arrow2/writer_async.rs

```diff
@@ -60,13 +49,24 @@ pub fn write_record_batches_to_stream(
         };
         (writable_stream, raw_stream.readable())
     };
-
+    
     spawn_local::<_>(async move {
+        let mut adapted_stream = batches.peekable();
+        let mut pinned_stream = std::pin::pin!(adapted_stream);
+        let first_batch = pinned_stream.as_mut().peek().await.unwrap();
+        let schema = first_batch.schema().into_inner();
+        // Need to create an encoding for each column
+        let mut encodings = vec![];
+        for _ in &schema.fields {
+            // Note, the nested encoding is for nested Parquet columns
+            // Here we assume columns are not nested
+            encodings.push(vec![encoding]);
+        }
```

Comment:
just a separate note that this was kinda a hack and at some point I need to come back and make sure that the encoding is compatible for each type in the schema